core/integration/directory_test.go

```diff
@@ -71,6 +72,48 @@ func TestDirectoryWithNewFile(t *testing.T) {
 	require.Equal(t, []string{"some-file"}, res.Directory.WithNewFile.Entries)
 }
 
+func TestDirectoryName(t *testing.T) {
+	t.Parallel()
+
+	wd := t.TempDir()
+
+	c, ctx := connect(t, dagger.WithWorkdir(wd))
+
+	t.Run("container dir", func(t *testing.T) {
+		dir := c.Container().From(alpineImage).Directory("/usr/bin")
+
+		name, err := dir.Name(ctx)
+		require.NoError(t, err)
+		require.Equal(t, "bin", name)
+	})
+
+	t.Run("container dir in dir", func(t *testing.T) {
+		file := c.Container().From(alpineImage).Directory("/usr").Directory("/bin")
+
+		name, err := file.Name(ctx)
+		require.NoError(t, err)
+		require.Equal(t, "bin", name)
+	})
+
+	t.Run("host dir", func(t *testing.T) {
+		err := os.MkdirAll(filepath.Join(wd, "dir"), 0o700)
+		require.NoError(t, err)
+
+		name, err := c.Host().Directory("dir").Name(ctx)
+		require.NoError(t, err)
+		require.Equal(t, ".", name)
```

Comment:
I'm not 100% sold on this being "correct".

But also, currently the top-level directory here isn't tracked, so we'd have to do that, thoughts?

Reply:
I think it would make more sense if it was `dir`, even if parents are not tracked in the context of the host.
Specially because if you do `c.Host().Directory(".")` which is the actual host root, what would be its name?